Fixup some database problems

fitzthum opened the pull request:
This should fix https://github.com/confidential-containers/simple-kbs/issues/64

Upgrading sqlx was not as simple as I would have liked. 

@ryansavino I am bumping the version of Rust which you pinned in our builder container a while ago. Do you think that's going to break things?

fitzthum committed:
versions: upgrade SQLx

Our old version of sqlx has some compatibility issues
with mariadb, so we've got to upgrade.

Upgrading requires some messing around due to upstream bugs.

Signed-off-by: Tobin Feldman-Fitzthum <tobin@ibm.com>

fitzthum committed:
docker: upgrade rust version

The new version of sqlx requires a new version of rust.

Signed-off-by: Tobin Feldman-Fitzthum <tobin@ibm.com>

fitzthum committed:
compose: remove deprecated versions tag

the versions field is no longer needed

Signed-off-by: Tobin Feldman-Fitzthum <tobin@ibm.com>

ryansavino commented:
The issue was the later bullseye images didn't support openssl v3 for some reason:
https://github.com/confidential-containers/simple-kbs/issues/61

1.80 may be okay though.

Also, docker hub images. Unfortunately, rust doesn't have a container image anywhere else besides docker hub.

@arvindskumar99 you have any additional insights here from your notes? I remember you having some issues recently with the simple-kbs.

fitzthum committed:
tools: update postgres db type

The variable set in this test script does not match
the one we use in the CI. Let's adjust.

Signed-off-by: Tobin Feldman-Fitzthum <tobin@ibm.com>

fitzthum committed:
db: fix policy insertion

Updating sqlx broke our policy insertion.

Unfortunately we have to adopt some backend-specific
logic.

Signed-off-by: Tobin Feldman-Fitzthum <tobin@ibm.com>

fitzthum commented:
Still ironing out some issues with upgrading sqlx. There were a lot of changes between v0.5.0 and v0.8.0

fitzthum committed:
db: add conditional decoding for MySQL

With the udpate to sqlx, we are no longer able
to transparently convert SQL BLOBS (such as longtext)
to Rust strings when using the mysql driver.

Add some branches to convert these into bytes first
and then to strings.

Signed-off-by: Tobin Feldman-Fitzthum <tobin@ibm.com>

fitzthum commented:
Ok, issues appear to be ironed out.
